src/components/TodoList/TodoItem.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import { observer } from 'mobx-react';
+import classNames from 'classnames';
+import useStore from '../../hooks/useStore';
+import { ITodo } from '../../types/todoTypes';
+
+const TodoItem = ({ content, completed, id }: ITodo) => {
+  const [isChecked, setIsChecked] = useState<boolean>(completed);
+  const [renderFlag, setRenderFlag] = useState<boolean>(true);
+  const [isEditing, setIsEditing] = useState<boolean>(false);
+  const [editedTodo, setEditedTodo] = useState<string>(content);
+  const { todoStore } = useStore();
+
+  useEffect(() => {
+    if (todoStore.path === '/completed') {
+      completed ? setRenderFlag(true) : setRenderFlag(false);
+    } else if (todoStore.path === '/active') {
+      completed ? setRenderFlag(false) : setRenderFlag(true);
+    } else {
+      setRenderFlag(true);
+    }
+  }, []);
+
+  const handleTodoChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setEditedTodo(value);
+  };
+  const handleTodoOnBlur = () => {
+    setIsEditing(!isEditing);
+  };
+  const handleKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
```

Comment:
```suggestion
    if (checkKeyBoardEnter(e.key)) {
```

아래 코드 부분과 추상화 단계를 맞춰주어서, 가독성을 높이는 것도 좋아보입니다.